pkg/storage/engine/mvcc.go

```diff
@@ -2728,21 +2728,25 @@ func MVCCFindSplitKey(
 	} else if !ok {
 		return nil, nil
 	}
-	minSplitKey := key
+	var minSplitKey roachpb.Key
 	if _, _, err := keys.DecodeTablePrefix(it.UnsafeKey().Key); err == nil {
 		// The first key in this range represents a row in a SQL table. Advance the
 		// minSplitKey past this row to avoid the problems described above.
 		firstRowKey, err := keys.EnsureSafeSplitKey(it.Key().Key)
 		if err != nil {
 			return nil, err
 		}
-		minSplitKey = roachpb.RKey(firstRowKey.PrefixEnd())
+		minSplitKey = firstRowKey.PrefixEnd()
```

Comment:
Wait, don't you want to drop the `.PrefixEnd` now that you compare against `minSplitKey` with `<` instead of `<=`?

Reply:
Oh, no, never mind. I confused myself. I do think that you're right that this is broken for interleaved tables.